### DISCUSSION TITLE
Output the radial stresses in spherical cordinates

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#23376] [Q&A General] [answered]
Hello MOOSE experts,

I am using the following module to calculate the stresses in the spherical coordinates.

https://mooseframework.inl.gov/source/kernels/StressDivergenceRSphericalTensors.html

Can you please suggest, how to get the output of stresses in spherical coordinate system? Is there any relevant examples?

Thanks,
Avtar

**Comment by hugary1995** [comment, marked as answer]
IIRC, if you use the axisymmetric RZ coordinate system in tensor mechanics, the strain and stress definitions are already in that coordinate system. The convention is
```
[ rr rz rt
  zr zz zt
  tr tz tt]
```
I suspect if you grab the corresponding component the same way as if you are in Cartesian, it will just work. But I agree this is a bit counter-intuitive, so hopefully there is a better way or it's been documented somewhere.

**Reply by sapitts** [reply]
In addition to the page you linked, there is the spherical strain documentation page with a bit more information [https://mooseframework.inl.gov/source/materials/ADComputeRSphericalFiniteStrain.html](https://mooseframework.inl.gov/source/materials/ADComputeRSphericalFiniteStrain.html). Note this formulation does assume a 1D problem

**Reply by hugary1995** [reply]
The new TM kernel system has a dedicated page for all that: https://mooseframework.inl.gov/source/interfaces/GradientOperator.html#2D_axisymmetric_cylindrical

Now that I look at it again, it is also somewhat incomplete...

**Reply by avtarsinghh1991** [reply]
Thanks. Its done.
